### PR TITLE
CFG and BasicBlock classes inititalized

### PR DESCRIPTION
**Pull request by ushakov-igor**

### CODE REVIEW

File: TACOptimizations/ControlFlowGraph.cs [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+            _basicBlocks = basicBlocks;
+        }
+
+        public List<BasicBlock> GetCurrentBasicBlocks() => _basicBlocks;
+
+        public List<BasicBlock> GetChildrenBasicBlock(int index)
+        {
+            var result = new List<BasicBlock>();
+            for (int j = 0; j < _children[index].Count; j++)
+                if (_children[index][j] == 1)
```
Comment (Molkree) [inline TACOptimizations/ControlFlowGraph.cs:29]: why 1?
how are _children/_parent arranged? are they just matrices of 0/1?

File: TACOptimizations/ControlFlowGraph.cs [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+                    result.Add(_basicBlocks[j]);
+            return result;
+        }
+
+        public List<BasicBlock> GetParentBasicBlock(int index)
+        {
+            var result = new List<BasicBlock>();
+            for (int j = 0; j < _children[index].Count; j++)
```
Comment (Molkree) [inline TACOptimizations/ControlFlowGraph.cs:37]: shouldn't it be _parent?

File: TACOptimizations/ControlFlowGraph.cs (hunk 1 above)
Comment (ushakov-igor) [reply TACOptimizations/ControlFlowGraph.cs:29]: https://en.wikipedia.org/wiki/Adjacency_matrix 
the guys who will build the graph said that for now let it be so.

File: TACOptimizations/ControlFlowGraph.cs (hunk 2 above)
Comment (ushakov-igor) [reply TACOptimizations/ControlFlowGraph.cs:37]: yep, my mistake

File: TACOptimizations/ControlFlowGraph.cs (hunk 1 above)
Comment (Molkree) [reply TACOptimizations/ControlFlowGraph.cs:29]: If it's simply an adjacency matrix why do you need two of them? Parents and children can be decoded in one matrix just fine.

Comment (Molkree) [reply TACOptimizations/ControlFlowGraph.cs:29]: Also `bool` is enough here, no need for `int`.

Comment (Molkree) [reply TACOptimizations/ControlFlowGraph.cs:29]: Refer me to those guys who said to use adj.matrix, please. S.S. Mihalkovich said to NOT store CFG as matrix but maybe I missed something? :)

Comment (ushakov-igor) [reply TACOptimizations/ControlFlowGraph.cs:29]: These guys promised to remake in any case to their needs, so this is a temporary.

**Pull request merged**